### PR TITLE
Internal - Set a min Artifactory version for Visibility System usage

### PR DESCRIPTION
**Pull request by eyalbe4**
* Set a minimum Artifactory version for the usage of the Visibility System.
* Remove the HTTP retries for requests sent to the Visibility System. 

Depends on https://github.com/jfrog/jfrog-client-go/pull/1063

### CODE REVIEW

**Review by EyalDelarea — APPROVED** [review]
Looks great !
Just see my comments on the client PR.

File: common/commands/command.go [hunk 1]
```diff
@@ -43,32 +50,45 @@ func reportUsage(command Command, channel chan<- bool) {
...
+		return
+	}
+	serviceManager, err := utils.CreateServiceManager(serverDetails, -1, 0, false)
+	if err != nil {
+		log.Debug("Usage reporting. Failed creating the Artifactory Service Manager.", err.Error())
+		return
+	}
+	ArtifactoryVersion, err := serviceManager.GetVersion()
+	if err != nil {
+		log.Debug("Usage reporting. Failed getting the Artifactory", err.Error())
```
Comment (sverdlov93) [inline common/commands/command.go:66]:
```suggestion
		log.Debug("Usage reporting: ", err.Error())
```
GetVersion errors are already wrapped and explaining the issue

File: common/commands/command.go [hunk 2]
```diff
@@ -43,32 +50,45 @@ func reportUsage(command Command, channel chan<- bool) {
...
+	}
+	if serverDetails == nil || serverDetails.ArtifactoryUrl == "" {
+		return
+	}
+	serviceManager, err := utils.CreateServiceManager(serverDetails, -1, 0, false)
+	if err != nil {
+		log.Debug("Usage reporting. Failed creating the Artifactory Service Manager.", err.Error())
+		return
+	}
+	ArtifactoryVersion, err := serviceManager.GetVersion()
```
Comment (sverdlov93) [inline common/commands/command.go:64]:
```suggestion
	artifactoryVersion, err := serviceManager.GetVersion()
```

File: go.mod [hunk 3]
```diff
@@ -96,7 +96,7 @@ require (
 	gopkg.in/warnings.v0 v0.1.2 // indirect
 )
 
-replace github.com/jfrog/jfrog-client-go => github.com/jfrog/jfrog-client-go v1.28.1-0.20241223175448-88f1089d0694
+replace github.com/jfrog/jfrog-client-go => github.com/eyalbe4/jfrog-client-go v1.28.1-0.20241225095445-ce46100f95eb
```
Comment (sverdlov93) [inline go.mod:99]: reminder to replace the eyalbe

**Pull request merged**